test/plugins/features/movies/index.test.js

```diff
@@ -23,6 +23,36 @@ describe('movies integration', () => {
 
   });
 
+  describe('location', () => {
+
+    it('associates a location', async () => {
+      const payload = { location: 'Berkeley' };
+
+      const response = await Movies.inject({
+        url: '/movies/1/locations',
+        method: 'POST',
+        payload
+      });
+
+      expect(response.statusCode).to.eql(200);
+      expect(response.result.object).to.eql('movie');
+      expect(response.result.locations).to.include.members(['Berkeley']);
+    });
+
+    it('does not associate a location multiple times', async () => {
+      const payload = { location: 'Berkeley' };
+
+      const response = await Movies.inject({
+        url: '/movies/1/locations',
+        method: 'POST',
+        payload
+      });
+
+      expect(response.statusCode).to.eql(500);
```

Comment:
I think it might be better to catch and throw a custom error (e.g. using [boom](https://github.com/lob/lob-api/blob/master/src/libraries/lobErrors.js)) instead of letting this 500, since this is behavior we are already anticipating. Totally up to you though.